browser-test/test.oauth2.ts

```diff
@@ -0,0 +1,223 @@
+    const opts = {
+      access_type: ACCESS_TYPE,
+      scope: SCOPE,
+      response_type: 'code token'
+    };
+
+    const generated = client.generateAuthUrl(opts);
+    const parsed = url.parse(generated);
+    if (typeof parsed.query !== 'string') {
+      throw new Error('Unable to parse querystring');
+    }
+    const query = qs.parse(parsed.query);
+    assert.strictEqual(query.response_type, 'code token');
+    assert.strictEqual(query.access_type, ACCESS_TYPE);
+    assert.strictEqual(query.scope, SCOPE);
+    assert.strictEqual(query.client_id, CLIENT_ID);
+    assert.strictEqual(query.redirect_uri, REDIRECT_URI);
+    done();
+  });
+
+  it('getToken should work', async () => {
+    const now = Date.now();
+    const stub =
+        sinon
+            .stub(client.transporter, 'request')
+            // @ts-ignore TS2345
+            .resolves({
+              data:
+                  {access_token: 'abc', refresh_token: '123', expires_in: 10}
+            });
+    client.getToken('code here', (err, tokens) => {
+      assert.isAbove(tokens!.expiry_date!, now + (10 * 1000));
+      assert.isBelow(tokens!.expiry_date!, now + (15 * 1000));
+    });
+  });
+
+  it('getFederatedSignonCerts talks to correct endpoint', async () => {
+    const stub = sinon
+                     .stub(client.transporter, 'request')
+                     // @ts-ignore TS2345
+                     .resolves(FEDERATED_SIGNON_JWK_CERTS_AXIOS_RESPONSE);
+    const result = await client.getFederatedSignonCertsAsync();
+    assert.strictEqual(result.format, CertificateFormat.JWK);
+    assert.strictEqual(
+        Object.keys(result.certs).length, FEDERATED_SIGNON_JWK_CERTS.length);
+    for (const cert of FEDERATED_SIGNON_JWK_CERTS) {
+      assert.deepStrictEqual(result.certs[cert.kid], cert);
+    }
+  });
+
+  it('getFederatedSignonCerts caches certificates', async () => {
+    const stub = sinon
+                     .stub(client.transporter, 'request')
+                     // @ts-ignore TS2345
+                     .resolves(FEDERATED_SIGNON_JWK_CERTS_AXIOS_RESPONSE);
+    const result1 = await client.getFederatedSignonCertsAsync();
+    const result2 = await client.getFederatedSignonCertsAsync();
+    assert(stub.calledOnce);
+    assert.deepStrictEqual(result1.certs, result2.certs);
+    assert.deepStrictEqual(result1.format, result2.format);
+  });
+
+  it('should generate a valid code verifier and resulting challenge',
+     async () => {
+       const codes = await client.generateCodeVerifierAsync();
+       assert.match(codes.codeVerifier, /^[a-zA-Z0-9\-\.~_]{128}$/);
+     });
+
+  it('should include code challenge and method in the url', async () => {
+    const codes = await client.generateCodeVerifierAsync();
+    const authUrl = client.generateAuthUrl({
+      code_challenge: codes.codeChallenge,
+      code_challenge_method: CodeChallengeMethod.S256
+    });
+    const parsed = url.parse(authUrl);
```

Comment:
`url.parse` is deprecated:
https://nodejs.org/docs/latest/api/deprecations.html#deprecations_dep0116_legacy_url_api

Please use the new WHATWG Url class

Reply:
no `URL` class in webpack yet: https://github.com/webpack/node-libs-browser/issues/69